flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/scala/expressionDsl.scala

```diff
@@ -1130,4 +1130,13 @@ object concat_ws {
   }
 }
 
+object log {
+  def apply(base: Expression, antilogarithm: Expression): Expression = {
+    Log(base, antilogarithm)
+  }
+  def apply(antilogarithm: Expression): Expression = {
+    new Log(antilogarithm)
```

Comment:
Is it just `new Ln(antilogarithm)`, we might be able to restructure them together?

Reply:
nvm, I think it might be too weird to write something like `base.log(antilogarithm)` in table API.

Reply:
1. I've tried it before, but it seems that we can't restructure log(x) and ln(x) together because the expression in sql will be reflected directly as an instance of case class. Please let me know if you figure it out.
2. Actually we should use log(base, antilogarithm).

Reply:
yup. I thought so too. Thanks for exploring all the possibilities 👍

Reply:
We should go for `antilog.log(base)` for consistency with `antilog.ln()` and `antilog.log10()`. 
Since we need to inverse the parameters of the case class, we need to adapt the `ExpressionParser`.